src/app/util/client-api.cpp

```diff
@@ -210,7 +210,8 @@ static uint16_t vFillBuffer(uint8_t * buffer, uint16_t bufferLen, uint8_t frameC
         // The destination buffer must be at least as large as the running total
         // plus the length of the integer value (if applicable) plus the length of
         // the data (if applicable).
-        if (bufferLen < bytes + dataLen + valueLen)
+        // Fix #19351 - Cast up to larger type during summation to prevent integer overflow issues.
+        if (bufferLen < (uint32_t) bytes + (uint32_t) dataLen + (uint32_t) valueLen)
```

Comment:
This is fine, but we should make https://github.com/project-chip/connectedhomeip/pull/19291 compile and merge it...